### PR TITLE
dae: init at 0.1.5

### PR DESCRIPTION
**Pull request by oluceps**
###### Description of changes
Add https://github.com/daeuniverse/dae

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/dae/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    unset STRIP
+    make CFLAGS="-D__REMOVE_BPF_PRINTK -fno-stack-protector" \
+    CLANG=${lib.getExe llvmPackages_15.clang} \
+    ebpf
+  '';
+
+  # network required
+  doCheck = false;
+
+  meta = with lib; {
```
Comment (NickCao) [inline pkgs/tools/networking/dae/default.nix:41]: Platform should be set to linux only.

File: pkgs/tools/networking/dae/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/daeuniverse/dae/cmd.Version=${version}"
+    "-X github.com/daeuniverse/dae/common/consts.MaxMatchSetLen_=64"
+  ];
+
+  preBuild = ''
+    unset STRIP
```
Comment (NickCao) [inline pkgs/tools/networking/dae/default.nix:32]: Or just use `NOSTRIP`? See https://github.com/daeuniverse/dae/blob/85343ac141768d4bcdc48a2ba9edff56b3b9c1a1/Makefile#LL16-L16C8

File: pkgs/tools/networking/dae/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    make CFLAGS="-D__REMOVE_BPF_PRINTK -fno-stack-protector" \
```
Comment (NickCao) [inline pkgs/tools/networking/dae/default.nix:33]: What are these `CFLAGS` for?

File: pkgs/tools/networking/dae/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    CLANG=${lib.getExe llvmPackages_15.clang} \
```
Comment (NickCao) [inline pkgs/tools/networking/dae/default.nix:34]:
```suggestion
    CLANG=${lib.getExe buildPackages.clang} \
```
(If the default version of clang works)

File: pkgs/tools/networking/dae/default.nix (hunk 3 above)
Comment (oluceps) [reply pkgs/tools/networking/dae/default.nix:32]: > What are these `CFLAGS` for?

Flag `-D__REMOVE_BPF_PRINTK` is for preventing [Unknown bpf_trace_printk error](https://github.com/daeuniverse/dae/blob/85343ac141768d4bcdc48a2ba9edff56b3b9c1a1/docs/getting-started/troubleshooting.md?plain=1#L11).

Compiler in NixOS force enabled stack protector which will causes compile fail. `-fno-stack-protector` for disabling it.

<details><summary>Log</summary>

```console
> nix build .#dae
error: builder for '/nix/store/qv3fxz1wv3mdxdf2zvzx0k2bba1kd6ig-dae-0.1.5.drv' failed with exit code 2;
       last 10 log lines:
       >     ^
       > /build/source/control/kern/tproxy.c:2173:5: error: A call to built-in function '__stack_chk_fail' is not supported.
       > int tproxy_wan_cg_sendmsg6(struct bpf_sock_addr *ctx) {
       >     ^
       > 11 errors generated.
       > Error: can't execute /nix/store/fam63d3x0fv1mb9i7igqikqzk1ngmcdx-clang-wrapper-11.1.0/bin/clang: exit status 1
       > exit status 1
       > control/control.go:8: running "go": exit status 1
       > make: *** [Makefile:51: ebpf] Error 1
       > /nix/store/sw36plhp82916wwg6i6097rkzza7d950-stdenv-linux/setup: line 136: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/qv3fxz1wv3mdxdf2zvzx0k2bba1kd6ig-dae-0.1.5.drv'.
```
</details>

File: pkgs/tools/networking/dae/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-EoStRmyYOtvG5ejtvHWNe9IIeE77hqp1OXBhRdxCYHs=";
+    fetchSubmodules = true;
+  };
+
+  vendorHash = "sha256-vxHufE3538l6zIcozFcrNhl+2sG1PtzkVxC0NyL3WMU=";
+
+  proxyVendor = true;
+
+  GOFLAGS = "-buildvcs=false";
```
Comment (NickCao) [inline pkgs/tools/networking/dae/default.nix:22]: `GOFLAGS` is shadowed by https://github.com/NixOS/nixpkgs/blob/a5aeb5016dec640058180411033f3c6409153646/pkgs/build-support/go/module.nix#L176
So it wasn't needed in the first place?

File: pkgs/tools/networking/dae/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/daeuniverse/dae/cmd.Version=${version}"
+    "-X github.com/daeuniverse/dae/common/consts.MaxMatchSetLen_=64"
+  ];
+
+  preBuild = ''
+    make CFLAGS="-D__REMOVE_BPF_PRINTK -fno-stack-protector" \
+    CLANG=${lib.getExe llvmPackages.clang} \
```
Comment (NickCao) [inline pkgs/tools/networking/dae/default.nix:33]: Or just add it to `nativeBuildInputs`, but cross compilation does not seem to work anyway.

File: pkgs/tools/networking/dae/default.nix (hunk 5 above)
Comment (oluceps) [reply pkgs/tools/networking/dae/default.nix]: > So it wasn't needed in the first place?

Seems yes.

**Review by NickCao — APPROVED** [review]

**Pull request merged**